Lib/typing.py

```diff
@@ -2395,7 +2381,7 @@ def __init__(self, name, tp):
         if '.' in name:
             name = name.rpartition('.')[-1]
         self.__name__ = name
-        self.__module__ = _callee(default='typing')
```

Comment:
I'm confused. Why don't we need this anymore?

Reply:
Looks like we should do it like it did at other classes:
```py
def_mod = _callee()
if def_mod != 'typing':
    self.__module__ = def_mod
```

This line was written by me and it can lead to pickling errors cause in case if name is not present it will set module name to `typing` and `typing` module can not contains this `NewType`.

What is your opinion regarding that?

Reply:
Wow interesting! Are you able to to reproduce this with a small peice of code? If you can, we should backport this specific change (and you should open a separate PR for this, tied to the pickling/serialization issue). We can't backport the current PR entirely because it is refactoring, not bugfix.

Reply:
Code to snippet to reproduce this issue:
```py
import pickle

from typing import NewType

del __name__

MyNewType = NewType("MyNewType", int)

pickle.dumps(MyNewType)
```

Reply:
Thanks. Please create a separate PR just for this and link to https://bugs.python.org/issue44353.

Reply:
New issue has been created.

link to issue - https://bugs.python.org/issue44761.
lint to PR - https://github.com/python/cpython/pull/27406